substrate/utils/frame/benchmarking-cli/src/storage/write.rs

```diff
@@ -64,14 +64,16 @@ impl StorageCmd {
 		let mut kvs: Vec<_> = trie.pairs(Default::default())?.collect();
 		let (mut rng, _) = new_rng(None);
 		kvs.shuffle(&mut rng);
-		info!("Writing {} keys", kvs.len());
+		let number_of_keys = (kvs.len() * self.params.db_fraction as usize) / 100;
+		let number_of_keys = if number_of_keys != 0 {number_of_keys} else {1};
 
+		info!("Writing {} keys of {} keys", number_of_keys, kvs.len());
 		let mut child_nodes = Vec::new();
 
 		// Generate all random values first; Make sure there are no collisions with existing
 		// db entries, so we can rollback all additions without corrupting existing entries.
-		for key_value in kvs {
-			let (k, original_v) = key_value?;
+		for key_value in kvs.iter().take(number_of_keys) {
+			let (k, original_v) = (key_value.clone())?;
```

Comment:
There should be no clone needed. Try `into_iter` maybe.

Reply:
You're right, I tested it with into_iter and it works.